docs/SetupTestEnvironment.md

```diff
@@ -0,0 +1,38 @@
+# Setup Test Environment
+
+## Create a Service Principal
+
+- Create a new Service Principal on Azure using [azure cli](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli?toc=%2Fen-us%2Fazure%2Fazure-resource-manager%2Ftoc.json&bc=%2Fen-us%2Fazure%2Fbread%2Ftoc.json&view=azure-cli-latest) or [azure portal](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-create-service-principal-portal?view=azure-cli-latest)
+- Assign the new Service Principal with [Contributor Role]
```

Comment:
If this SP is for ACR, the scope should be the ACR resource, not the entire resource group

Reply:
SP isn't required for ACR.  However, following the guidance here through to enabling Admin user & access keys - allows you then to use username & password displayed within the CI/CD pipeline. eg YAML extract:

```
- script: |
    docker build -f Dockerfile -t $(AzureContainerRegistry)/$(ContainerName):$(Build.BuildId) .
    docker login -u $(AcrUser) -p $(AcrPassword) $(AzureContainerRegistry)
    docker push $(AzureContainerRegistry)/$(ContainerName):$(Build.BuildId)
  displayName: 'Build and push Docker image'
```
https://docs.microsoft.com/en-us/azure/container-registry/container-registry-get-started-portal#create-a-container-registry